Common/Orders/TimeInForce.cs

```diff
@@ -0,0 +1,159 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using QuantConnect.Interfaces;
+using QuantConnect.Orders.TimeInForces;
+using QuantConnect.Securities;
+
+namespace QuantConnect.Orders
+{
+    /// <summary>
+    /// Time In Force - defines the length of time over which an order will continue working before it is canceled
+    /// </summary>
+    public abstract class TimeInForce : IEquatable<TimeInForce>, ITimeInForceHandler
+    {
+        /// <summary>
+        /// Gets a <see cref="TimeInForce"/> instance with <see cref="TimeInForceType.GoodTilCanceled"/> type
+        /// </summary>
+        public static readonly TimeInForce GoodTilCanceled = new GoodTilCanceledTimeInForce();
+
+        /// <summary>
+        /// Gets a <see cref="TimeInForce"/> instance with <see cref="TimeInForceType.Day"/> type
+        /// </summary>
+        public static readonly TimeInForce Day = new DayTimeInForce();
+
+        /// <summary>
+        /// The type of Time In Force
+        /// </summary>
+        public TimeInForceType Type { get; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TimeInForce"/> class
+        /// </summary>
+        protected TimeInForce(TimeInForceType type)
+        {
+            Type = type;
+        }
+
+        /// <summary>
+        /// Converts a <see cref="TimeInForceType"/> value to a new <see cref="TimeInForce"/> instance.
+        /// </summary>
+        public static implicit operator TimeInForce(TimeInForceType type)
+        {
+            switch (type)
+            {
+                case TimeInForceType.GoodTilCanceled:
+                    return new GoodTilCanceledTimeInForce();
+
+                case TimeInForceType.Day:
+                    return new DayTimeInForce();
```

Comment:
Instead of creating a new object, can we use the existing `static readonly` instances already defined? IIRC, all of these supported implementations are stateless.

Reply:
Sure, this was an oversight